internal/analysis/dolos/engine.go

```diff
@@ -44,8 +62,28 @@ func (this *dolosEngine) IsAvailable() bool {
 	return docker.CanAccessDocker()
 }
 
-func (this *dolosEngine) ComputeFileSimilarity(paths [2]string, templatePath string, ctx context.Context) (*model.FileSimilarity, error) {
-	err := ensureImage()
+func parseEngineOptions(rawOptions model.OptionsMap) (*DolosEngineOptions, error) {
+	effectiveOptions := GetDefaultDolosOptions()
+
+	if rawOptions == nil {
+		return effectiveOptions, nil
+	}
+
+	effectiveOptions, err := util.JSONTransformTypes(rawOptions, effectiveOptions)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to convert raw options to DolosEngineOptions: '%w'.", err)
+	}
+
+	return effectiveOptions, nil
+}
+
+func (this *dolosEngine) ComputeFileSimilarity(paths [2]string, templatePath string, ctx context.Context, options model.OptionsMap) (*model.FileSimilarity, error) {
+	effectiveOptions, err := parseEngineOptions(options)
```

Comment:
Let's rename "options" so it's more clear that people shouldn't be using it pass the first line of this function,
and instead they should be using "effectiveOptions".

Maybe:
s/options/rawOptions/
s/effectiveOptions/options/